File: functions-introduction/solutions/city-weather-node/methods.js
const path = require('path');
var fs = require("fs");


module.exports = {

   read_weather(cityName) {
      const filePath = path.resolve(__dirname, "cities.json");

      if (!fs.existsSync(filePath)) {
          console.error('File not found'+filename);
          process.exit();
      }

      const filecontents = fs.readFileSync(filePath, "utf8");
      const weather = JSON.parse(filecontents);

      const cityLowerCase = (cityName || "").toLowerCase()
      return weather.city[cityLowerCase];
   },

   kelvin_to_celsius(temp_kelvin) {
      return temp_kelvin - 273.15;
   },

   kelvin_to_farhenheit(temp_kelvin) {
      return temp_kelvin * 1.8 - 459.67;
   }
}

File: functions-introduction/apps/resources/methods.py
from pathlib import Path
import json

def read_weather(city: str):
    weather = {}
    filepath = Path(__file__).parent.joinpath("cities.json")

    try:
        with open(filepath) as data_file:
            weather = json.load(data_file)

    except FileNotFoundError as e:
        print("No cities.json file exists!!")
        raise e

    cityname = (city or "").lower()
    city = weather['city'].get(cityname)

    return city


def kelvin_to_celsius(temperature: float):
    return temperature - 273.15


def kelvin_to_farhenheit(temperature: float):
    return temperature * 1.8 - 459.67
